packages/in_app_purchase/example/lib/main.dart

```diff
@@ -374,13 +374,14 @@ class _MyAppState extends State<MyApp> {
             _handleInvalidPurchase(purchaseDetails);
           }
         }
-        if (Platform.isIOS) {
-          InAppPurchaseConnection.instance.completePurchase(purchaseDetails);
-        } else if (Platform.isAndroid) {
+        if (Platform.isAndroid) {
           if (!kAutoConsume && purchaseDetails.productID == _kConsumableId) {
             InAppPurchaseConnection.instance.consumePurchase(purchaseDetails);
           }
         }
+        if (purchaseDetails.pendingCompletePurchase) {
+          InAppPurchaseConnection.instance.completePurchase(purchaseDetails);
+        }
```

Comment:
This should only be done if the product is valid and we've delivered it, right? It looks like the block above is missing an early return under `_handleInvalidPurchase`.

Reply:
added `return;` to both `_handleInvalidPurchase` and `handleError`